Packs/CrowdStrikeFalconX/Integrations/CrowdStrikeFalconX/CrowdStrikeFalconX.py

```diff
@@ -0,0 +1,693 @@
+        params = {
+            "filter": filter,
+            "offset": offset,
+            "limit": limit,
+            "sort": sort,
+        }
+        return self._http_request("Get", url_suffix, params=params)
+
+    def find_submission_id(
+            self,
+            limit: int,
+            filter: str,
+            offset: str,
+            sort: str,
+    ) -> Union[dict, str, Response]:
+        """Creating the needed arguments for the http request
+        :param limit: maximum number of report IDs to return
+        :param filter: optional filter and sort criteria in the form of an FQL query
+        :param offset: the offset to start retrieving reports from.
+        :param sort: sort order: asc or desc
+        :return: http response
+        """
+        url_suffix = f"/falconx/queries/submissions/v1?filter={filter}&offset={offset}&limit{limit}=&sort={sort}"
+
+        params = {
+            "filter": filter,
+            "offset": offset,
+            "limit": limit,
+            "sort": sort,
+        }
+        return self._http_request("Get", url_suffix, params=params)
+
+
+def add_outputs_from_dict(
+        api_current_dict: dict,
+        fields_to_keep: list
+) -> dict:
+    """
+    Filters a dict and keeps only the keys that appears in the given list
+    :param api_current_dict: the origin dict
+    :param fields_to_keep: the list which contains the wanted keys
+    :return: a dict based on api_current_dict without the keys that doesn't appear in fields_to_keep
+    """
+    if not api_current_dict or not fields_to_keep:
+        return {}
+
+    group_outputs = {}
+
+    for field_to_keep in fields_to_keep:
+        if field_to_keep in api_current_dict.keys():
+            group_outputs[field_to_keep] = api_current_dict.get(field_to_keep)
+
+    return group_outputs
+
+
+def parse_outputs(
+        api_res: Dict,
+        meta_fields: list = [],
+        quota_fields: list = [],
+        resources_fields: list = [],
+        sandbox_fields: list = []
+) -> Dict[str, dict]:
+    """Parse group data as received from CrowdStrike FalconX API into Demisto's conventions
+    the output from the API is a dict that contains the keys: meta, resources and errors
+    the meta contains a "quota" dict
+    the "resources" is an array that contains the sandbox dict
+    the function filters the wanted params from the api result
+    :param api_res: the api result from the http request
+    :param meta_fields: the wanted params that appear in the mate section
+    :param quota_fields: the wanted params that appear in the quota section
+    :param resources_fields: the wanted params that appear in the resources section
+    :param sandbox_fields: the wanted params that appear in the sandbox section
+    :return: a dict based on api_res with the wanted params only
+    """
+    if api_res.get("errors"):
+        # if there is an error in the api result, return only the error
+        return {"errors": api_res.get("errors")[0]}
```

Comment:
what happens if there is more than one error in the list?
let's do a utility func for this and make sure we unit testts all the options:
no "erorrs", errors is a string/list with 1 item/multiple items/empty list

Reply:
couldn't create a use-case that causes more than 1 error, checking with Elior

Reply:
No need to re create via API, just make sure you can handle it in the code if it will happen.